floodsub.go

```diff
@@ -14,18 +14,26 @@ const (
 	FloodSubID = protocol.ID("/floodsub/1.0.0")
 )
 
+// New returns a new floodsub-enabled PubSub objecting using the protocols specified in ps
+func New(ctx context.Context, h host.Host, ps []protocol.ID, opts ...Option) (*PubSub, error) {
```

Comment:
This API is motivated by two things:

1) I didn't want to introduce a breaking change to your existing API
2) I didn't think that it was quite the time to significantly refactor the mechanism for configuring the FloodSub (to match what was recently incorporated into the DHT, for example)

Reply:
So, given that we have both a FloodSub and a PubSub, I'd rather not call this `New` (a bit ambiguous`). It's a mouthful but how about `NewFloodsubWithProtocols`. Eventually, we can deprecate this but at least it won't clash with any names we may need in the future. For example, we may want to define a `New` constructor in the future that sets up the default pubsub routers.

Reply:
Sounds totally reasonable to me. Just made the change and pushed.